File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    answer = list(my_string)
    for i in queries:
        a = []
        for n in range(i[0], i[1]+1):
            a.insert(0,answer[n])
        for idx in range(len(a)):
            answer[idx + i[0]] = a[idx]
    return ''.join(answer)

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        char[] letters = my_string.toCharArray();
        for (int[] index : queries) {
            int start = index[0];
            int end = index[1];
            while (start < end) {
                char temp = letters[start];
                letters[start] = letters[end];
                letters[end] = temp;
                start++;
                end--;
            }
        }
        String answer = "";
        for (char c : letters) {
            answer += c;
        }
        return answer;
    }
}